v2/common/src/main/java/com/google/cloud/teleport/v2/transforms/ElasticsearchTransforms.java

```diff
@@ -0,0 +1,259 @@
+   *       (Default:5000000).
+   *   <li>{@link WriteToElasticsearchOptions#getMaxRetryAttempts()} - optional: maximum retry
+   *       attempts for {@link ElasticsearchIO.RetryConfiguration}.
+   *   <li>{@link WriteToElasticsearchOptions#getMaxRetryDuration()} - optional: maximum retry
+   *       duration for {@link ElasticsearchIO.RetryConfiguration}.
+   *   <li>{@link WriteToElasticsearchOptions#getUsePartialUpdate()} - use partial updates instead
+   *       of insertions (Default: false).
+   * </ul>
+   */
+  @AutoValue
+  public abstract static class WriteToElasticsearch extends PTransform<PCollection<String>, PDone> {
+
+    public static Builder newBuilder() {
+      return new AutoValue_ElasticsearchTransforms_WriteToElasticsearch.Builder();
+    }
+
+    public abstract WriteToElasticsearchOptions options();
+
+    @Override
+    public PDone expand(PCollection<String> jsonStrings) {
+
+      ConnectionConfiguration config =
+          ConnectionConfiguration.create(
+              options().getNodeAddresses().split(","),
+              options().getIndex(),
+              options().getDocumentType());
+
+      RetryConfiguration retryConfig = null;
+      if (options().getMaxRetryAttempts() != null) {
+        retryConfig =
+            RetryConfiguration.create(
+                options().getMaxRetryAttempts(), getDuration(options().getMaxRetryDuration()));
+      }
+
+      ElasticsearchIO.Write write =
+          ElasticsearchIO.write()
+              .withConnectionConfiguration(config)
+              .withMaxBatchSize(options().getBatchSize())
+              .withMaxBatchSizeBytes(options().getBatchSizeBytes())
+              .withUsePartialUpdate(options().getUsePartialUpdate());
+
+      if (retryConfig != null) {
+        write.withRetryConfiguration(retryConfig);
+      }
+
+      return jsonStrings.apply(
+          "WriteDocuments",
+          write
+              .withIdFn(
+                  ValueExtractorFn.newBuilder()
+                      .setFileSystemPath(options().getIdFnPath())
+                      .setFunctionName(options().getIdFnName())
+                      .build())
+              .withIndexFn(
+                  ValueExtractorFn.newBuilder()
+                      .setFileSystemPath(options().getIndexFnPath())
+                      .setFunctionName(options().getIndexFnName())
+                      .build())
+              .withTypeFn(
+                  ValueExtractorFn.newBuilder()
+                      .setFileSystemPath(options().getTypeFnPath())
+                      .setFunctionName(options().getTypeFnName())
+                      .build()));
+    }
+
+    /** Builder for {@link WriteToElasticsearch}. */
+    @AutoValue.Builder
+    public abstract static class Builder {
+      public abstract Builder setOptions(WriteToElasticsearchOptions options);
+
+      public abstract WriteToElasticsearch build();
```

Comment:
You need to validate all the options inside of an implementation for a build method.

Reply:
Implemented build method for validation of options.